addons/mail/static/src/components/composer_text_input/composer_text_input.scss

```diff
@@ -20,7 +20,11 @@
     resize: none;
     border-radius: $o-mail-rounded-rectangle-border-radius-lg;
     border: none;
-    overflow: auto;
+    overflow-x: hidden;
+
+    &:placeholder-shown {
+        white-space: nowrap;
+    }
```

Comment:
I think what you want is to apply this ruleset instead:

```scss

&::placeholder {
    white-space: nowrap;
    overflow-x: hidden;
    text-overflow: ellipsis;
}
```

Reply:
Good find, that might be just what we need.

Reply:
Well actually, this is close to the first solution I tested, but the ellipsis doesn't work.
The result look the same, but with `::placeholder` we can move the overflow. I think it's a better solution then.

Reply:
At least I've tested on Safari and `text-overflow: ellipsis` works as expected with my suggestion.

Reply:
Yeah... For some reasons, only Safari does the desired `text-overflow: ellipsis;`. It doesn't work on Chrome and Firefox...

I don't think we can solve this problem on current versions of Chrome/Firefox with just CSS. I wouldn't attempt solving it with JS code.